lib/Doctrine/Migrations/Configuration/Loader/XmlFileLoader.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Migrations\Configuration\Loader;
+
+use Doctrine\Migrations\Configuration\Configuration;
+use Doctrine\Migrations\Configuration\Exception\FileNotFound;
+use Doctrine\Migrations\Configuration\Exception\XmlNotValid;
+use Doctrine\Migrations\Tools\BooleanStringFormatter;
+use DOMDocument;
+use SimpleXMLElement;
+use const DIRECTORY_SEPARATOR;
+use const LIBXML_NOCDATA;
+use function assert;
+use function file_exists;
+use function file_get_contents;
+use function libxml_clear_errors;
+use function libxml_use_internal_errors;
+use function simplexml_load_string;
+use function strtr;
+
+/**
+ * @internal
+ */
+final class XmlFileLoader extends AbstractFileLoader
+{
+    /** @var ArrayLoader */
+    private $arrayLoader;
+
+    public function __construct(?ArrayLoader $arrayLoader = null)
```

Comment:
It's obviously a hard dependency, what do you think about making that clear by removing both the

- nullable type declaration and the
- default value

?